lib/active_record/connection_adapters/sqlserver/database_statements.rb

```diff
@@ -198,7 +198,12 @@ def sql_for_insert(sql, pk, id_value, sequence_name, binds)
           end
           sql = if pk && self.class.use_output_inserted && !database_prefix_remote_server?
                   quoted_pk = SQLServer::Utils.extract_identifiers(pk).quoted
-                  sql.insert sql.index(/ (DEFAULT )?VALUES/), " OUTPUT INSERTED.#{quoted_pk}"
+                  if !self.class.use_output_inserted_into
+                    sql.dup.insert sql.index(/ (DEFAULT )?VALUES/), " OUTPUT INSERTED.#{quoted_pk}"
+                  else
+                    tempsql = sql.insert sql.index(/ (DEFAULT )?VALUES/), " OUTPUT INSERTED.#{quoted_pk} INTO @inserted_id"
+                    "DECLARE @inserted_id table (id int);#{tempsql}; SELECT * FROM @inserted_id;"
```

Comment:
* Assume `use_output_inserted` uses `INTO`. No need for two formats.
* Change `(id int)` to something like `CAST(#{quoted_pk} AS bigint)`